src/pybind/mgr/volumes/fs/subvolume.py

```diff
@@ -189,6 +189,28 @@ def get_subvolume_path(self, spec):
             raise VolumeException(-e.args[0], e.args[1])
         return path
 
+    def get_dir_entries(self, path):
+        """
+        Get the directory names in a given path
+        :param path: the given path
+        :return: the list of directory names in json format
+        """
+        try:
+            dir_handle = self.fs.opendir(path)
+        except cephfs.ObjectNotFound:
+            raise VolumeException(-errno.ENOENT,
+                                  "Path '{0}' not found".format(path))
+        except cephfs.Error as e:
+            raise VolumeException(-e.args[0], e.args[1])
+
+        dirs = []
+        d = self.fs.readdir(dir_handle)
+        while d:
+            dirs.append({'name': d.d_name.decode('utf-8')})
```

Comment:
You've got half of the presentation code in this subvolume module. Paths should remain byte objects except when presented to the user. The appropriate place to transform these paths into a JSON serializable object is only in `volume.py`.

I would suggest returning a list of `d_name`s. Then create a helper method in `volume.py` that changes it into a serializeable JSON object.

Reply:
I thought that was simple/straight-forward. It needs a new function then. I'm fixing it now.